### PR TITLE
Add RFCs for Commands and Model API

### PR DESCRIPTION
**Pull request by sr229**

### CODE REVIEW

**Comment by sr229** [comment]
CC @LeNitrous, its best if you do the implementation details for the Model API, I only covered the basic lifecycle details.

File: client/2022/March/rfc-commands-api.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+### Allowed types in extension API
+
+Vignette does not impose restrictions on which data to transfer between extensions, and as such, most data is allowed, which can be any but not limited to the following:
+
+- `Blob`
+- `Object`
+- `Number`
+- `String`
+- `ReadableStream`
```
Comment (LeNitrous) [inline client/2022/March/rfc-commands-api.md:67]: We can allow any kind of Javascript data type as no parsing is needed when commands are dispatched programmatically.

**Review by LeNitrous — CHANGES_REQUESTED** [review]
As far as the models API is concerned, they should only render and update models. I'm not sure why they also have to send a rigging provider. Pretty sure that can be handled internally by the extension itself as many common model systems have a skeleton system of their own.

If the intent was to allow lets say moving a part of the user's body move a different skeleton on the model, then I think this should not be in extension level.

File: client/2022/March/rfc-commands-api.md (hunk 1 above)
Comment (sr229) [reply client/2022/March/rfc-commands-api.md:67]: I think I already said this since I said "..can be any but not limited to the following".

Comment (LeNitrous) [reply client/2022/March/rfc-commands-api.md]: I think it reads better when you explicitly say "any Javascript data type".

**Review by LeNitrous — APPROVED** [review]
I'd want to hear more about the skeleton system you're planning but this works for now.

**Comment by sr229** [comment]
Oh shoot I don't think the last patch made it ヽ(*。>Д<)o゜

**Pull request merged**